pkg/controllers/deprovisioning/types.go

```diff
@@ -69,28 +75,38 @@ func NewCandidate(ctx context.Context, kubeClient client.Client, clk clock.Clock
 	instanceTypeMap := provisionerToInstanceTypes[node.Labels()[v1alpha5.ProvisionerNameLabelKey]]
 	// skip any nodes where we can't determine the provisioner
 	if provisioner == nil || instanceTypeMap == nil {
+		recorder.Publish(deprovisioningevents.Blocked(node.Node, node.Machine, fmt.Sprintf("owning provisioner %q not found", provisioner.Name))...)
```

Comment:
More dry to publish outside of NewCandidate?

Reply:
We _could_ do it that way but we'd have to create an error that differentiates between a recoverable and non-recoverable state. Tell me if there is another way that you are thinking about this, but I think we need the context of some of the details in this function (beyond just the error message) to know whether to fire an event or not